tests/forte/processors/data_augment/algorithms/word_splitting_processor_test.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for Random Word Splitting Data Augmentation processor
+"""
+
+import unittest
+import random
+
+from forte.data.selector import AllPackSelector
+from forte.pipeline import Pipeline
+from forte.data.multi_pack import MultiPack
+from forte.data.readers import StringReader
+from forte.data.caster import MultiPackBoxer
+from forte.processors.data_augment.algorithms.word_splitting_processor import (
+    RandomWordSplitDataAugmentProcessor,
+)
+
+from forte.processors.misc import WhiteSpaceTokenizer
+from ft.onto.base_ontology import Token
+
+from ddt import ddt, data, unpack
+
+
+@ddt
+class TestWordSplittingProcessor(unittest.TestCase):
+    def setUp(self):
+        random.seed(5)
+        self.nlp = Pipeline[MultiPack]()
+
+        boxer_config = {"pack_name": "input_src"}
+
+        self.nlp.set_reader(reader=StringReader())
+        self.nlp.add(component=MultiPackBoxer(), config=boxer_config)
+        self.nlp.add(
+            component=WhiteSpaceTokenizer(), selector=AllPackSelector()
+        )
+
+    @data(
+        (
+            [
+                "Mary and Samantha arrived at the bus station early "
+                "but waited until noon for the bus ."
+            ],
```

Comment:
Could we try an two-sentence data. and then use `PeriodSentenceSplitter`, see if the resulting `Sentence` is aligned as expected?